lib/Parser/Parse.cpp

```diff
@@ -8443,6 +8474,19 @@ ParseNodePtr Parser::ParseExpr(int oplMin,
             // Parse the operand, make a new node, and look for more
             pnodeT = ParseExpr<buildAST>(opl, NULL, fAllowIn, FALSE, pNameHint, &hintLength, &hintOffset, nullptr);
 
+            // Detect nested function escapes of the pattern "o.f = function(){...}" or "o[s] = function(){...}".
+            // Doing so in the parser allows us to disable stack-nested-functions in common cases where an escape
+            // is not detected at byte code gen time because of deferred parsing.
+            if (m_currentNodeFunc && pnodeT && pnodeT->nop == knopFncDecl && fIsDotOrIndex && nop == knopAsg)
```

Comment:
Not necessarily. It will be non-null if we're enclosed in a non-deferred function, regardless of whether the current (possibly nested) function is deferred. If we're not enclosed in a non-deferred function, we won't generate any stack-nested-function byte code.